packages/transaction-controller/src/TransactionController.ts

```diff
@@ -898,6 +902,50 @@ export class TransactionController extends BaseController<
     this.incomingTransactionHelper.stop();
   }
 
+  /**
+   * Adds external provided transaction to state as confirmed transaction.
+   *
+   * @param transactionMeta - TransactionMeta to add transactions.
+   * @param transactionReceipt - TransactionReceipt of the external transaction.
+   * @param baseFeePerGas - Base fee per gas of the external transaction.
+   */
+  async confirmExternalTransaction(
+    transactionMeta: TransactionMeta,
```

Comment:
Do we need the falsey check that the extension has to return if the `transactionMeta` is not set?

Reply:
Since `transactionMeta` passed over to `validateConfirmedExternalTransaction` I've added check in validation function, didn't wanted to do those checks here and there tbh